### DISCUSSION TITLE
Some questions after switching from 12L

### DISCUSSION THREAD

**Discussion by sanityseeker** [Archfx/duo-de#58] [Q&A] [answered]
Hi! I have a couple of questions regarding the launcher and the system in general. Would be grateful for your help here.

1. What is the expected behavior in a tent mode? There seems to be an inconsistency where the launcher initially presents the landscape view (when tend mode is used) but then reverts to portrait mode. As a result, the Duo is standing in a tent mode with the launcher reverted to portrait. Despite this, the system acknowledges the preference for landscape orientation in tent mode, as evidenced by applications running in landscape (e.g. youtube and others). Yet, the launcher persists in portrait orientation.

Maybe there is some setting about allowing the home screen rotation that I'm missing?

2. Launcher settings are not preserved after an OTA. I installed the latest image yesterday, but still it shows in settings that the update is available. After downloading it and rebooting, all apps organization was lost :(

3. I guess there is no way to force apps opening on one side of the screen as stock duo did? I think I'll get used to it, but damn that Microsoft behavior seems superior now for that duo format when I lost it :)

4. There is a space for 6 favorite icons in "landscape/tablet" mode, but only 5 are allowed in "portrait/phone" mode. Not a huge issue, but the 6th icon is always disappearing after switching back from portrait to landscape. I guess we can call it a bug.

5. When using desktop mode with windows, is there a way to tile the window to a half of one screen or send to another screen. I know that there are keyboard shortcuts for this in a normal non-desktop mode. If not, maybe you could share your use-cases for the current implementation of a desktop mode?

6. Do you know if it is possible or not to design custom keyboard shortcuts to switch between applications, or to launch a specific application? For example, in split-screen, it seems to be impossible to switch focus between apps without touching the screen.

All in all, thank you greatly for supporting this beautiful hardware and prolonging its life! It is so exciting and fun to try new things on the Duo. That's a pity that the Google's stock desktop mode implementation is still so far from the Samsung's quality( Smth like Dex could really suit this amazing portable 2-screen hardware.

**Comment by Archfx** [comment, marked as answer]
Hi,

Answering your questions,

> 1. What is the expected behavior in a tent mode? There seems to be an inconsistency where the launcher initially presents the landscape view (when tend mode is used) but then reverts to portrait mode. As a result, the Duo is standing in a tent mode with the launcher reverted to portrait. Despite this, the system acknowledges the preference for landscape orientation in tent mode, as evidenced by applications running in landscape (e.g. youtube and others). Yet, the launcher persists in portrait orientation.

Actually, this is not an issue. In phone mode, default settings do not rotate the screen with auto-rotation. You have to enable it from the `home settings`. When you are in Phone mode, click and hold on the display, click `Home settings`, and enable the feature `Allow home screen rotation`. This will enable the launcher to serve the Tent mode and Ramp modes correctly. Read me is updated with this information.

> 2. Launcher settings are not preserved after an OTA. I installed the latest image yesterday, but still it shows in settings that the update is available. After downloading it and rebooting, all apps organization was lost :(

This is not an issue with the OTA, it was an issue in the previous release, the launcher was resetting on the reboots. The latest release (v20), fixed it. What you are experiencing is the unsaved state from the previous release. I have this information in the release notes.


> 3. I guess there is no way to force apps opening on one side of the screen as stock duo did? I think I'll get used to it, but damn that Microsoft behavior seems superior now for that duo format when I lost it :)

Unfortunately, I don't plan to incorporate this since this causes lot of issues with the default AOSP launcher (Trebuchet). In Thai's initial release, he switched to the Pixel launcher to fix this, but I just like to keep the pure Android experience as it is.

> 4. There is a space for 6 favorite icons in "landscape/tablet" mode, but only 5 are allowed in "portrait/phone" mode. Not a huge issue, but the 6th icon is always disappearing after switching back from portrait to landscape. I guess we can call it a bug.

This needs to be checked since the duo hardware is not a traditional implementation of tablet and phone, it requires huge reverse engineering work to even make it usable. This is such an instance for such example.

> 5. When using desktop mode with windows, is there a way to tile the window to a half of one screen or send to another screen. I know that there are keyboard shortcuts for this in a normal non-desktop mode. If not, maybe you could share your use-cases for the current implementation of a desktop mode?

This is an interesting idea, window management is done by the Android system. Actually, the current implementation does not overlap the windows, this logic can be modified. But it takes a lot of time, I don't know if it will be worth the effort. 


> 6. Do you know if it is possible or not to design custom keyboard shortcuts to switch between applications, or to launch a specific application? For example, in split-screen, it seems to be impossible to switch focus between apps without touching the screen.

I see where you are coming from. I am also a huge fan of sway and tiling windows. This should be coming from the functionality of the keyboard side. Unfortunately, I am not an Android app developer, if somebody modifies the default keyboard app logic to launch apps, I will definitely consider incorporating it into the system.

> All in all, thank you greatly for supporting this beautiful hardware and prolonging its life! It is so exciting and fun to try new things on the Duo. That's a pity that the Google's stock desktop mode implementation is still so far from the Samsung's quality( Smth like Dex could really suit this amazing portable 2-screen hardware.

Thanks, I appreciate it. I like the concept of Duo, but the concept brings a lot of challenges in making it usable. That is why even the Stock 12L had a lot of issues. That is also a reason for third-party launchers not working well on Duo devices. At the end of the day, it is hard for me to do a lot of reverse engineering stuff, and fix stuff alone.  It is a LOT of WORK. That is one of the reasons why I made everything public.

**Reply by sanityseeker** [reply]
Thank you!

I have one additional question if you don't mind. It is regarding the OTA updates.
The build number is the following `treble_arm64_bgN-userdebug 14 AP2A.240805.005 eng.aruna.20240920.134435`.
Updater shows that there is a new version, downloads it, applies, and after the reboot everything stays the same. Still shows that there is a new version available, the build number doesn't change. If I understand correctly, this build number should correspond to the latest version already. Is this behavior expected?
![Screenshot_20240923-185506](https://github.com/user-attachments/assets/956ed5d4-eb51-42a3-9beb-ba4f57b2849e)

**Reply by Archfx** [reply]
I am removing this concern. Future OTA updates will not show the same version as a new update available.

**Comment by Archfx** [comment]
I have updated the answers for Q2 and to the followup question.
